packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -263,18 +263,28 @@ class AndroidDevice extends Device {
   }
 
   bool _installLatestApp(ApplicationPackage package) {
-    if (isAppInstalled(package)) {
+    final bool wasInstalled = isAppInstalled(package);
+    if (wasInstalled) {
       if (isLatestBuildInstalled(package)) {
         printStatus('Latest build already installed.');
         return true;
       }
-      printStatus('Uninstalling old version...');
-      if (!uninstallApp(package))
-        printError('Warning: uninstalling old version failed');
     }
     printTrace('Installing APK.');
     if (!installApp(package)) {
-      printTrace('Error: Failed to install APK.');
+      printStatus('Warning: Failed to install APK.');
```

Comment:
i think this should be a trace, otherwise when it's successful the only messages you see are:
```
Warning: Failed to install APK.
Uninstalling old version...
```
...which sounds like failure.